src/osd/PrimaryLogPG.cc

```diff
@@ -1082,14 +1081,14 @@ void PrimaryLogPG::do_pg_op(OpRequestRef op)
 
   int result = 0;
   string cname, mname;
-  PGLSFilter *filter = NULL;
   bufferlist filter_out;
 
   snapid_t snapid = m->get_snapid();
 
   vector<OSDOp> ops = m->ops;
 
   for (vector<OSDOp>::iterator p = ops.begin(); p != ops.end(); ++p) {
+    std::unique_ptr<PGLSFilter> filter;
```

Comment:
That's the key change in the commit. I'm not sure whether the carrying behavior is intended or not. This PR is just an excuse to ignite discussion.